app/javascript/components/visualization/controls/ScatterPlotLegend.js

```diff
@@ -203,6 +291,32 @@ export default function ScatterPlotLegend({
           </>
           }
         </div>
+        { enableColorPicking &&
+          <div>
+            { showColorControls &&
+              <>
+                <span>Click a label to select a new color</span><br/>
+                <div>
+                  <a role="button" data-analytics-name="legend-color-picker-save" onClick={saveColors}>
+                    Save colors
+                  </a>
+                  <a role="button" className="pull-right" data-analytics-name="legend-color-picker-cancel" onClick={cancelColors}>
+                    Cancel
+                  </a><br/>
+                  &nbsp;
+                  <a role="button" className="pull-right" data-analytics-name="legend-color-picking-reset" onClick={resetColors}>
```

Comment:
Blocker: sorry, my last suggestion missed this:

```suggestion
                  <a role="button" className="pull-right" data-analytics-name="legend-color-picker-reset" onClick={resetColors}>
```